skill-tree/src/main/java/com/RDS/skilltree/Endorsement/EndorsementServiceImpl.java

```diff
@@ -59,4 +64,38 @@ public EndorsementModel createEndorsement(EndorsementDRO endorsementDRO) {
             throw new NoEntityException("Skill with id:" + skillId + " not found");
         }
     }
+
+    @Override
+    public GenericResponse<Void> updateEndorsementStatus(String id, String status) {
+        UserModel user =
+                (UserModel) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+        if (!user.getRole().equals(UserRole.SUPERUSER)) {
+            throw new AccessDeniedException("Unauthorized access");
+        }
+        if (!(Objects.equals(status, EndorsementStatus.APPROVED.name())
+                || Objects.equals(status, EndorsementStatus.REJECTED.name()))) {
+            throw new IllegalArgumentException("Invalid endorsement status: " + status);
+        }
+
+        try {
+            UUID endorsementId = UUID.fromString(id);
+            Optional<EndorsementModel> optionalEndorsementModel =
+                    endorsementRepository.findById(endorsementId);
+            if (optionalEndorsementModel.isPresent()) {
+                EndorsementModel updatedEndorsementModel =
+                        EndorsementModel.builder()
+                                .id(optionalEndorsementModel.get().getId())
+                                .user(optionalEndorsementModel.get().getUser())
+                                .skill(optionalEndorsementModel.get().getSkill())
+                                .endorsersList(optionalEndorsementModel.get().getEndorsersList())
+                                .status(EndorsementStatus.valueOf(status))
+                                .build();
+                endorsementRepository.save(updatedEndorsementModel);
```

Comment:
As discussed on Discord, If this throws any exception it would be DataAccessException which is ultimately a subclass of Exception. So, it will go to the `globalExceptionHandler`. We don’t have to handle it separately.

Reply:
Ok, resolving this. Just double check and verify once so we are sure.